provider/cache.go

```diff
@@ -74,9 +83,33 @@ func (c *cached[T]) Get() (T, error) {
 func (c *cached[T]) Reset() {
 	c.mux.Lock()
 	c.updated = time.Time{}
+	c.retried = time.Time{}
 	c.mux.Unlock()
 }
 
 func (c *cached[T]) mustUpdate() bool {
-	return c.clock.Since(c.updated) > c.cache || errors.Is(c.err, api.ErrMustRetry)
+	return (c.clock.Since(c.updated) > c.cache && c.err == nil) || errors.Is(c.err, api.ErrMustRetry) || c.shouldRetryWithBackoff(c.err)
+}
+
+func (c *cached[T]) shouldRetryWithBackoff(err error) bool {
+	if err != nil {
+		// Exponentially backoff for 2^n minutes. Maximum backoff wait time is regular cache time
+		waitTime := time.Duration(math.Min(math.Pow(2, float64(c.backoffCounter)), c.cache.Seconds())*60) * time.Second
```

Comment:
Wollen wir vllt. bei 1s anfangen? Das wären immer noch wenige Retries und kein DOS.

Reply:
Das lässt sich dann aber schwer exponentiell abbilden, wenn die restlichen Schritte 2^n Minuten sein sollen.

Reply:
Brauchts das denn?

Reply:
Was?

Reply:
Dass die Schritte in Minuten sind? Wozu? Passt eh nie zum interval?

Reply:
Was schwebt dir vor, wie es stattdessen aussehen sollte?

Reply:
Ich hätte vllt. mal mit 5s angefangen (oder auch noch weniger). Bei 10s Intervall führt das dazu, dass es erstmal 2 Versuche gibt was kein Problem sein sollte. Danach gehts dann hoch.

Reply:
Also exponentiell zur Basis 5 in Sekunden? Wären dann 5^0=1, 5^1=5, 5^2=25, 5^3=125, 5^4=625. Also beim Default-Cache von 15 Minuten 5 Versuche und dann knapp 5 Minuten warten.

Reply:
Wenn man erst mit 5 Sekunden starten will, könnte man auch den Counter bei 1 starten und hätte dann einen Versuch weniger. Aber ich glaub fast, 5 Versuche sollten ok sein.